diracx-routers/src/diracx/routers/utils/users.py

```diff
@@ -0,0 +1,101 @@
+import re
+from typing import Annotated, Any
+from uuid import UUID
+
+from authlib.jose import JoseError, JsonWebToken
+from fastapi import Depends, HTTPException, status
+from fastapi.security import OpenIdConnect
+from pydantic import BaseModel, Field
+
+from diracx.core.models import UserInfo
+from diracx.core.properties import SecurityProperty
+from diracx.core.settings import FernetKey, ServiceSettingsBase, TokenSigningKey
+from diracx.routers.dependencies import add_settings_annotation
+
+oidc_scheme = OpenIdConnect(openIdConnectUrl="/.well-known/openid-configuration")
+
+
+class AuthInfo(BaseModel):
+    # raw token for propagation
+    bearer_token: str
+
+    # token ID in the DB for Component
+    # unique jwt identifier for user
+    token_id: UUID
+
+    # list of DIRAC properties
+    properties: list[SecurityProperty]
+
+    policies: dict[str, Any] = {}
+
+
+class AuthorizedUserInfo(AuthInfo, UserInfo):
+    pass
+
+
+@add_settings_annotation
+class AuthSettings(ServiceSettingsBase, env_prefix="DIRACX_SERVICE_AUTH_"):
+    """Settings for the authentication service."""
+
+    dirac_client_id: str = "myDIRACClientID"
+    # TODO: This should be taken dynamically
+    # ["http://pclhcb211:8000/docs/oauth2-redirect"]
+    allowed_redirects: list[str] = []
+    device_flow_expiration_seconds: int = 600
+    authorization_flow_expiration_seconds: int = 300
+
+    # State key is used to encrypt/decrypt the state dict passed to the IAM
+    state_key: FernetKey
+
+    token_issuer: str = "http://lhcbdirac.cern.ch/"
```

Comment:
I guess this should be mandatory? If not, we should at least have:

```suggestion
    token_issuer: str = "https://my-issuer.invalid/"
```

Reply:
added the todo